File: September-4-2025/containsDuplicate.js
// Contains Duplicate
// Solved 
// Given an integer array nums, return true if any value appears more than once in the array, otherwise return false.

// Example 1:

// Input: nums = [1, 2, 3, 3]

// Output: true

// Example 2:

// Input: nums = [1, 2, 3, 4]

// Output: false

var nums = [1, 2, 3, 3]

const containsDuplicate = (nums) => {
  var set = new Set()

  for(let num of nums){ 
    if(set.has(num)){ 
      return true
    } else { 
      set.add(num)
    }
  }
  return false
} 

console.log(containsDuplicate(nums))

File: 2025/November-7-2025/containsDuplicate.py
from typing import List

nums = [1, 2, 3, 4]

def containsDuplicate(nums: List[int]) -> bool: 
  numbers = set()

  for num in nums: 
    if num in numbers: 
      return True
    else: 
      numbers.add(num)
  return False

print(containsDuplicate(nums))